src/components/scenes/CreateWalletCompletionScene.tsx

```diff
@@ -88,28 +71,49 @@ const CreateWalletCompletionComponent = (props: Props) => {
   // Create the wallets and enable the tokens
   useAsyncEffect(
     async () => {
-      const promises: Array<(() => Promise<EdgeCurrencyWallet>) | (() => Promise<void>)> = [...walletPromises]
-      if (tokenKey != null) promises.push(tokenPromise)
+      let tokenPromise: Promise<void> | undefined
+      if (tokenKey != null) {
+        tokenPromise = dispatch(enableTokensAcrossWallets(newTokenItems)).then(
+          () => setItemStatus(currentState => ({ ...currentState, [newTokenItems[0].key]: 'complete' })),
+          error => {
+            showError(error)
+            setItemStatus(currentState => ({ ...currentState, [newTokenItems[0].key]: 'error' }))
+          }
+        )
+      }
+      const walletResults = await createWallets(
+        account,
+        newWalletItems.map(
+          (item): EdgeCreateCurrencyWallet => ({
+            fiatCurrencyCode: `iso:${fiatCode}`,
+            importText,
+            keyOptions: keyOptions.get(item.pluginId),
+            name: walletNames[item.key],
+            walletType: item.walletType
+          })
+        )
+      )
 
-      for (const [i, promise] of promises.entries()) {
-        try {
-          const wallet = await promise()
-          // We created a wallet so let's Update relevant pending tokens with the new walletId
+      for (let i = 0; i < walletResults.length; ++i) {
+        const result = walletResults[i]
+        if (!result.ok) {
+          showError(result.error)
+          setItemStatus(currentState => ({ ...currentState, [filteredCreateItemsForDisplay[i].key]: 'error' }))
+        } else {
+          const wallet = result.result
+          // We created a wallet so let's update relevant pending tokens with the new walletId
           if (wallet != null) {
             newTokenItems
               .filter(item => item.pluginId === wallet.currencyInfo.pluginId && item.createWalletIds[0] === PLACEHOLDER_WALLET_ID)
               .forEach(item => (item.createWalletIds = [wallet.id]))
           }
           setItemStatus(currentState => ({ ...currentState, [filteredCreateItemsForDisplay[i].key]: 'complete' }))
-        } catch (e) {
-          showError(e)
-          setItemStatus(currentState => ({ ...currentState, [filteredCreateItemsForDisplay[i].key]: 'error' }))
         }
-
         flatListRef.current?.scrollToIndex({ animated: true, index: i, viewPosition: 0.5 })
       }
-      setDone(true)
 
+      if (tokenPromise != null) await tokenPromise
```

Comment:
It may look weird to continue using scrollToIndex in the for/loop since it should execute very quickly. Maybe scroll after the for loop and again after the token promise? Or remove it and just let the scene be scrollable instead

Reply:
I like the idea of letting the scene be scrollable. I'll do that.